### DISCUSSION TITLE
What is the correct strategy and device for mps

### DISCUSSION THREAD

**Discussion by sneiman** [Lightning-AI/pytorch-lightning#15858] [General]
Trying to get some baby stuff working with mps - but keep getting the error:

`RuntimeError: Mismatched Tensor types in NNPack convolutionOutput`
This occurs as nn.Conv2d() is passed the data during a forward() call.

The sign on from ptl shows mps selected:
```
GPU available: True (mps), used: True
TPU available: False, using: 0 TPU cores
IPU available: False, using: 0 IPUs
HPU available: False, using: 0 HPUs
```
But the data at entry to forward shows device as -1.

I've tried lots of different settings for strategy and device. Any help appreciated.

**Comment by alejandroarmas** [comment]
Do you have an update on your situation? I'm experiencing the same thing

**Reply by Jonathan-Nyquist** [reply]
I'm also having this problem.

**Comment by nathom** [comment]
The issue for me was mixing CPU tensors with GPU tensors. If `x` is the input to `Conv2d`, ensure you call `x.to(device)` before passing it in if the convolution is running on your GPU.
